src/nvim/search.c

```diff
@@ -1137,7 +1137,7 @@ int do_search(
 
       // Get the offset, so we know how long it is.
       if (spats[0].off.line || spats[0].off.end || spats[0].off.off) {
-        p = off_buf;
```

Comment:
`p` starts pointing to `off_buff` here but stops pointing at line 1158, as it gets re-assigned to a different value